watertap/core/plugins/solvers.py

```diff
@@ -0,0 +1,139 @@
+        #       each time based on the initial values, just like in Ipopt.
+        try:
+            constraint_autoscale_large_jac(self._model,
+                    ignore_constraint_scaling=ignore_constraint_scaling,
+                    ignore_variable_scaling=ignore_variable_scaling,
+                    max_grad=max_grad,
+                    min_scale=min_scale)
+        except AssertionError as err:
+            if str(err) == "Error in AMPL evaluation":
+                print("ipopt-watertap: Issue in AMPL function evaluation; Jacobian constraint scaling not applied.")
+                halt_on_ampl_error = self.options.get("halt_on_ampl_error", "yes")
+                if halt_on_ampl_error == "no" :
+                    print("ipopt-watertap: halt_on_ampl_error=no, so continuing with optimization.")
+                else:
+                    raise RuntimeError("Error in AMPL evaluation.\n"
+                            "Run ipopt with halt_on_ampl_error=yes and symbolic_solver_labels=True to see the affected function.")
+            else:
+                print("Error in constraint_autoscale_large_jac")
+                raise
+
+        # this creates the NL file, among other things
+        super()._presolve(*args, **kwds)
+
+    def _postsolve(self):
+        self._reset_scaling_factors()
+        # remove our reference to the model
+        del self._model
+        return super()._postsolve()
+
+    def _cache_scaling_factors(self):
+        self._scaling_cache = [ (c, get_scaling_factor(c)) for c in
+                self._model.component_data_objects(
+                    pyo.Constraint, active=True, descend_into=True)]
+
+    def _reset_scaling_factors(self):
+        for c, s in self._scaling_cache:
+            if s is None:
+                unset_scaling_factor(c)
+            else:
+                set_scaling_factor(c, s)
+        del self._scaling_cache
+
+    def _get_option(self, option_name, default_value):
+        # NOTE: options get reset to their original value at the end of the
+        #       OptSolver.solve. The options in _presolve (where this is called)
+        #       are already copies of the original, so it is safe to pop them so
+        #       they don't get sent to Ipopt.
+        option_value = self.options.pop(option_name, None)
+        if option_value is None:
+            option_value = default_value
+        else:
+            if self._tee:
+                print(f"ipopt-watertap: {option_name}={option_value}")
+        return option_value
+
+    def _set_user_scaling(self):
+        if "nlp_scaling_method" in self.options and self.options["nlp_scaling_method"] != "user-scaling":
+            _log.warning("The ipopt-watertap solver is designed to be run with user-scaling. "
+                    f"Provided nlp_scaling_method {self.options['nlp_scaling_method']} will be ignored")
+        self.options["nlp_scaling_method"] = "user-scaling"
```

Comment:
So it looks like using this new solver will prevent a user from requesting 'gradient-based' scaling, which is what IDAES generally does. 

(1) Are we all ok with this behavior? 
(2) Will we still be able to update 'bound_push' and other args as needed (even in future PRs on this)?

Reply:
For cases where the user wants `gradient-based` scaling, they should probably use the standard IPOPT solver instead of this one. That said, for simplicity of workflows, it might make sense to allow users to specify gradient based scaling here and just skip the additional steps (just so we don't have to explain when to use one solver over the other).

Reply:
@andrewlee94 that's an interesting alternative. I think we should go with it.

Reply:
@aladshaw3 to answer your question (2): the only options that are not passed are those directly related to `gradient-based`, which have analogous in the IDAES Jacobian autoscaling tool (`nlp_scaling_max_gradient` and `nlp_scaling_min_value`). When Ipopt isn't using `gradient-based` scaling, these options are ignored anyways.

Reply:
Theoretically the the gradient based scaling provided by IDAES is exactly the same as the gradient scaling that Ipopt does, so it shouldn't make a difference if you use this or gradient scaling in Ipopt.  I did some testing when the IDAES Jacobian scaling was in early development and confirmed that it was exactly the same, but probably wouldn't hurt to reconfirm that and write some tests along those lines.

What using gradient scaling in ipopt does do, is make Ipopt ignore user-provided scale factors.